static/app/components/guidedSteps/guidedSteps.tsx

```diff
@@ -60,7 +63,26 @@ function useGuidedStepsContentValue({
 }: Pick<GuidedStepsProps, 'onStepChange'>): GuidedStepsContextState {
   const registeredStepsRef = useRef<RegisteredSteps>({});
   const [totalSteps, setTotalSteps] = useState<number>(0);
-  const [currentStep, setCurrentStep] = useState<number>(1);
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const currentStep = decodeInteger(location.query.guidedStep);
```

Comment:
@priscilawebdev this component was built to be unopinionated and flexible, so I disagree with coupling it to a specific query param.

I'd suggest adding an `initialStepId` prop and using the `onStepChange()` callback to update the query param in your onboarding component.

Reply:
@malwilley I completely understand and can definitely decouple this. However, considering that this component will be used throughout our product, it could lead to a frustrating user experience if the step state is lost every time the URL is refreshed

Reply:
At this point, it seems that in most places where we're using this feature, we'll need to include something like:

```
initialStep={decodeInteger(location.query.guidedStep)}
onStepChange={step => {
  navigate({
    pathname: location.pathname,
    query: {
      ...location.query,
      guidedStep: step,
    },
  });
}}
```

Are there any scenarios you can think of where this wouldn’t be the case?

Reply:
What I like about the `initialStep` approach is that we could use other ways of saving the current step, like localStorage or an API response depending on the need. In the future if it looks like we are copying this code a bunch perhaps we can create a more opinionated `<OnboardingSteps />` component or something like that that uses GuidedSteps internally?

Reply:
I see... that makes sense! We'll probably have to create another component, since we're copying and pasting in at least 3x different places. We'll see :)